### PR TITLE
Add native dynasm backend

### PR DESCRIPTION
**Pull request by takikawa**
This PR adds a dynasm backend for pflua. I'd appreciate any code review or comments, but here are some specific design points that I could use some feedback on:

* Switching to the native backend uses a `native = true` key/value pair in the table passed to `pf.compile`. Is this a good idea or is there a better interface? Should native be default? (my feeling is no, since it is less mature)
* I used environment variables in some places to enable native mode for tests (e.g., in pfquickcheck). Does that sound reasonable?
* Currently targets just x64 with a unix ABI. Are there any other ABIs/ISAs that should definitely be supported? (either way it's probably a good idea to output a useful error though---or should it fall back to Lua?)

Also if I missed any tests that I should convert to also test the dynasm backend that would be good to know.

The main files that were added are `pf/codegen.dasl`, `pf/regalloc.lua`, and `pf/selection.lua`. I plan to squash commits before merging. An additional submodule was added for the Lua port of dynasm.

### CODE REVIEW

File: doc/l1.md [hunk 1]
```diff
@@ -72,3 +72,41 @@ end
...
+7fbe1e3d201d  7F44              jg 0x7fbe1e3d2063
+7fbe1e3d201f  0FB7770E          movzx esi, word [rdi+0xe]
+7fbe1e3d2023  4881FEFEFE0000    cmp rsi, 0xfefe
+7fbe1e3d202a  7537              jnz 0x7fbe1e3d2063
+7fbe1e3d202c  0FB67711          movzx esi, byte [rdi+0x11]
+7fbe1e3d2030  4881FE83000000    cmp rsi, 0x83
+7fbe1e3d2037  752A              jnz 0x7fbe1e3d2063
+7fbe1e3d2039  0FB67715          movzx esi, byte [rdi+0x15]
+7fbe1e3d203d  4883FE0F          cmp rsi, +0x0f
+7fbe1e3d2041  7502              jnz 0x7fbe1e3d2045
```
Comment (wingo) [inline doc/l1.md]: Looks like you can tighten up this code -- insead of jnz + jmp you can jz + fallthrough, no?

Comment (takikawa) [inline doc/l1.md]: Yes, you're right. I just pushed a commit that implements that behavior (was planning to implement this earlier but didn't make it at the time I made the PR).

**Comment by wingo** [comment]
LGTM.  I find the code really tasteful!  Just one question about whether the .md files are up to date.  Excellent work Asumu!

**Comment by wingo** [comment]
lgtm, let's do this!!!

File: doc/l1.md (hunk 1 above)
Comment (takikawa) [inline doc/l1.md]: Yep, you're right. I just pushed an update.

**Comment by takikawa** [comment]
This is now squashed and merged as 4fbb6b509596ae616bcca747f8b6cc2d7d0a0d00 and ab88cf95d3945ee2c46516bc7f64acb96235b1d9.

**Pull request closed without merging**